src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js

```diff
@@ -11,35 +11,94 @@ const CheckYourEmailTitle = View.extend({
     'data-se': 'o-form-explain',
   },
   template: hbs`
-    {{i18n code="oie.email.verify.alternate.sentText" bundle="login"}}
+    {{#if email}}
+      {{i18n 
+          code="oie.email.verify.alternate.magicLinkToEmailAddress" 
+          bundle="login" 
+          arguments="email" 
+          $1="<span class='strong'>$1</span>"
+      }}
+    {{else}}
+      {{i18n 
+        code="oie.email.verify.alternate.magicLinkToYourEmail" 
+        bundle="login" 
+        $1="<span class='strong'>$1</span>"
+      }}
+    {{/if}}
+    {{i18n 
+      code="oie.email.verify.alternate.instructions" 
+      bundle="login" 
+    }}
   `,
 
   getTemplateData() {
     const { email } = this.options;
-    return {
-      email,
-    };
+    return { email };
   },
 });
 
-const Body = BaseAuthenticatorEmailForm.extend(Object.assign({
+const EnterCodeLink = View.extend({
+  template: hbs`
+    <button 
+      class="button-link enter-auth-code-instead-link"
+    >
+        {{i18n code="oie.email.verify.alternate.showCodeTextField"}}
+    </button>
+  `,
+});
 
-  resendEmailAction: 'currentAuthenticatorEnrollment-resend',
+const Body = BaseAuthenticatorEmailForm.extend(
+  Object.assign({
+    noButtonBar: true,
+    resendEmailAction: 'currentAuthenticatorEnrollment-resend',
 
-  initialize() {
-    BaseAuthenticatorEmailForm.prototype.initialize.apply(this, arguments);
+    initialize() {
+      BaseAuthenticatorEmailForm.prototype.initialize.apply(this, arguments);
 
-    const { email } = this.options.currentViewState.relatesTo?.value?.profile || {};
+      const { email } =
+        this.options.currentViewState.relatesTo?.value?.profile || {};
 
-    this.add(CheckYourEmailTitle, {
-      prepend: true,
-      selector: '.o-form-error-container',
-      options: {
-        email,
-      }
-    });
-  },
-}));
+      this.add(EnterCodeLink, {
+        prepend: true,
+        selector: '.o-form-error-container',
+      });
+
+      this.add(CheckYourEmailTitle, {
+        prepend: true,
+        selector: '.o-form-error-container',
+        options: { email },
+      });
+    },
+
+    postRender() {
+      this.showCodeEntryField(false);
+    },
+
+    events: {
```

Comment:
i would raise `events` above `initialize`

Reply:
Are they supposed to be in alphabetical order?

Reply:
No.. just all functions together and ideally `initialize` defined first.